analytic_engine/src/sst/parquet/writer.rs

```diff
@@ -511,6 +518,49 @@ impl<'a> SstWriter for ParquetSstWriter<'a> {
     }
 }
 
+struct ColumnEncodingSampler<'a> {
+    sample_row_groups: &'a [RecordBatchWithKey],
+    meta_data: &'a MetaData,
+    min_num_sample_rows: usize,
+    max_unique_value_ratio: f64,
+}
+
+impl<'a> ColumnEncodingSampler<'a> {
```

Comment:
This sampler is not necessary here, we already keep distinct values for low cardinality column in sst metadata, we can use them directly.

When pick sst for compaction,  we can get metadata of input sst, then we can decide which column need dict encoding there directly.

Reply:
For the sst whose level > 0, the stats about the column values in the metadata can be used (I'll fix it in a following PR). However, level-0 ssts don't contain such information so let's keep it here.